src/platform/iotjs_module_adc-linux-general.inl.h

```diff
@@ -16,28 +16,95 @@
 #ifndef IOTJS_MODULE_ADC_LINUX_GENERAL_INL_H
 #define IOTJS_MODULE_ADC_LINUX_GENERAL_INL_H
 
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
 
+#include "iotjs_systemio-linux.h"
 #include "module/iotjs_module_adc.h"
 
 
+#define ADC_INTERFACE "/sys/devices/12d10000.adc/iio:device0/"
+#define ADC_PIN_INTERFACE "in_voltage%d_raw"
+#define ADC_PIN_FORMAT ADC_INTERFACE ADC_PIN_INTERFACE
+
+#define ADC_PATH_BUFFER_SIZE DEVICE_IO_PATH_BUFFER_SIZE
+#define ADC_PIN_BUFFER_SIZE DEVICE_IO_PIN_BUFFER_SIZE
+#define ADC_VALUE_BUFFER_SIZE 64
+
+
+// Implementation used here are based on:
+//  https://www.kernel.org/doc/Documentation/adc/sysfs.txt
+
+
+void InitializeadcWorker(uv_work_t* work_req) {
+  iotjs_adc_reqwrap_t* req_wrap = iotjs_adc_reqwrap_from_request(work_req);
+  iotjs_adc_reqdata_t* req_data = iotjs_adc_reqwrap_data(req_wrap);
+  DDDLOG("ADC InitializeadcWorker()");
+
+  // Check if ADC interface exits.
+  if (DeviceCheckPath(ADC_INTERFACE)) {
+    req_data->result = kAdcErrOk;
+  } else {
+    req_data->result = kAdcErrExport;
+  }
+}
+
+void ReadadcWorker(uv_work_t* work_req) {
+  iotjs_adc_reqwrap_t* req_wrap = iotjs_adc_reqwrap_from_request(work_req);
+  iotjs_adc_reqdata_t* req_data = iotjs_adc_reqwrap_data(req_wrap);
+  uint32_t pin = req_data->pin;
+  DDDLOG("adc ReadadcWorker() - pin: %d", pin);
+
+  char value_path[ADC_PATH_BUFFER_SIZE] = { 0 };
+  snprintf(value_path, sizeof(value_path), ADC_PIN_FORMAT, pin);
+
+  char buffer[ADC_VALUE_BUFFER_SIZE];
+  if (DeviceOpenReadClose(value_path, buffer, sizeof(buffer))) {
+    req_data->result = kAdcErrOk;
+    req_data->value = atoi(buffer);
+  } else {
+    req_data->result = kAdcErrRead;
+  }
+}
+
+
+
 void iotjs_adc_export_worker(uv_work_t* work_req) {
-  IOTJS_ASSERT(!"Not implemented");
+  InitializeadcWorker(work_req);
 }
 
 
 void iotjs_adc_read_worker(uv_work_t* work_req) {
-  IOTJS_ASSERT(!"Not implemented");
+  ReadadcWorker(work_req);
 }
 
 
 int32_t iotjs_adc_read_sync(int32_t pin) {
-  IOTJS_ASSERT(!"Not implemented");
+  char path[ADC_PATH_BUFFER_SIZE];
+  snprintf(path, sizeof(path), ADC_PIN_INTERFACE, pin);
+  FILE* file = fopen(path, "r");
+  if (NULL == f) {
+    return -1;
+  }
+  int32_t value;
+  if (1 == fscanf(f, "%d", &value)) {
+    fclose(file);
```

Comment:
As we are closing the file in both cases, maybe a different sequence could be used. Something like this? (pseudo-code-ish):
```
read_count = fscanf(...)
fclose(..)
if (read_count == 1) { return  value }
return -1
```
What do you think?